### PR TITLE
Don't infer 'Object?' in an null coalescing expression.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/4483

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -1009,9 +1009,7 @@ private IEnumerable<ITypeSymbol> InferTypeInCatchFilterClause(CatchFilterClauseS
                 var rightTypes = GetTypes(coalesceExpression.Right);
                 if (!rightTypes.Any())
                 {
-                    return SpecializedCollections.SingletonEnumerable(
-                        this.Compilation.GetSpecialType(SpecialType.System_Nullable_T)
-                            .Construct(Compilation.GetSpecialType(SpecialType.System_Object)));
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Wow, we really tried hard to do the wrong thing.

**Comment by Pilchie** [comment]
:+1:

**Comment by dpoeschl** [comment]
:+1:

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Yeah.  I was expecting some weird confluence of events to have causes this.  Little did i expect :)

**Comment by CyrusNajmabadi** [comment]
Fixes #4483

**Pull request merged**